### PR TITLE
DAS-1451 - Implement HOSS-MaskFill chained service.

### PR DESCRIPTION
**Pull request by owenlittlejohns**
## Jira Issue ID

DAS-1451

## Description

This PR adds a chained services that combines the Harmony OPeNDAP SubSetter (HOSS) with MaskFill. This will be used to allow polygon spatial subsetting of geographically gridded data hosted in OPeNDAP. HOSS extracts a minimally encompassing bounding box, and retrieves those data from OPeNDAP. The output is then passed to MaskFill which masks pixels within the bounding box, but outside the polygon.

Another minor addition in this PR is to specify that the Trajectory Subsetter operations includes the `spatialSubset` option. The omission of this entry from the list was preventing shape file information reaching the Trajectory Subsetter.

### To test the HOSS-MaskFill chain:

_This might not work - I've been seeing issues with OPeNDAP hanging for RSSMIF16D requests. I think this relates to HYRAX-744._

* Configure the new service entry to run with RSSMIF16D:

```yaml
    collections:
      - C1222931739-GHRC_CLOUD
```

* Pull the DAS-1451-RSSMIF16D-configuration branch from the [MaskFill repository](https://git.earthdata.nasa.gov/projects/SITC/repos/maskfill). Run the `bin/build-harmony-image` script.
* Pull the DAS-1451 branch from the [HOSS repository](https://git.earthdata.nasa.gov/projects/SITC/repos/var_subsetter/browse). Run the `bin/build-image` script.
* Reload services.yml: `bin/reload-services-config`.
* Restart the services in your local Harmony instance: `bin/restart-services`.
* Make a shape file request for RSSMIF16D (note - you'll need to restrict your request to the ocean in the eastern hemisphere):

```python
from harmony import Client, Collection, Environment, Request

harmony_client = Client(env=Environment.LOCAL)
rssmif16d = Collection(id='C1222931739-GHRC_CLOUD')
request = Request(collection=rssmif16d,
                  shape='/path/to/local/shape.geo.json',
                  max_results=1)
```

### To test Trajectory Subsetter updates:

* Clone the [Trajectory Subsetter](https://git.earthdata.nasa.gov/projects/SITC/repos/trajectorysubsetter/browse). Build the latest Docker image using `bin/build-image`.
* Reload `services.yml`: `bin/reload-services-config`.
* Restart Harmony services: `bin/restart-services`.
* Make a shape file request for GEDI L4A (there's currently 1 test granule available, so Egypt is a great shape file):

```python
from harmony import Client, Collection, Environment, Request

gedi_l4a = Collection(id='C1242267295-EEDTEST')
egypt_granule_id = 'G1242274852-EEDTEST'
polygon_subset_request = Request(collection=gedi_l4a, granule_id=[egypt_granule_id],
                                 variables=variables, shape='/path/to/egypt.geo.json')
polygon_subset_job_id = harmony_client.submit(polygon_subset_request)
```

## PR Acceptance Checklist
* [x] Acceptance criteria met
* [x] Tests added/updated (if needed) and passing
* [x] Documentation updated (if needed)

### CODE REVIEW

**Comment by owenlittlejohns** [comment]
The error in the build is due to some vulnerabilities in NPM dependencies that haven't been changed in this PR. I can look into these tomorrow (Tues 31st).

**Comment by owenlittlejohns** [comment]
I've rebased this branch to contain the updates @chris-durbin made to fix the issues found by `npm audit`.

File: config/services.yml [hunk 1]
```diff
@@ -983,3 +983,34 @@ https://cmr.uat.earthdata.nasa.gov:
...
+        dimension: true
+        shape: true
+        variable: true
+      output_formats:
+        - application/netcdf # Incorrect mime-type, remove when no longer needed
+        - application/x-netcdf4
+    steps:
+      - image: !Env ${QUERY_CMR_IMAGE}
+      - image: !Env ${VAR_SUBSETTER_IMAGE}
+        operations: ['variableSubset', 'spatialSubset', 'dimensionSubset', 'shapefileSubset']
```
Comment (chris-durbin) [inline config/services.yml:1014]: Should the variable subsetter receive the shape information or should only the maskfill image receive the shape?

**Review by chris-durbin — COMMENTED** [review]
Once we get our deployments sorted out I'll test this change locally. There's one part of our code that I wasn't 100% sure was ready to support `shapefileSubset` in the list of operations.

**Review by flamingbear — APPROVED** [review]
I pulled this and the associated PRs related to DAS-1451 and was able to follow the instructions to test both the HOSS-maskfill and trajectory subsetter tests.

**Pull request merged**